### PR TITLE
Add support for dust template engine

### PR DESCRIPTION
**Pull request by hacksparrow**
This PR adds support for dust template engine https://github.com/expressjs/generator/issues/90.

The implementation is a little different from other template engines due to the fact that we have to use [`dustjs-linkedin`](https://www.npmjs.com/package/dustjs-linkedin), instead of `dust` (abandoned) or `dustjs`.

`dustjs-linkedin` does not provide a `__express()`-like method for rendering the templates, so we have to use a wrapper module, [`adaro`](https://www.npmjs.com/package/adaro).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 temp/
 npm-debug.log
+.DS_Store
```
Comment (dougwilson) [inline .gitignore]: This is something that should be in your global `.gitignore`, not in each and every project's. Please remove.

File: README.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ Install dependencies:
 $ npm install
 ```
 
-Rock and Roll
+Start the app:
```
Comment (dougwilson) [inline README.md]: Please use a different PR for this change, rather than bundling with an unrelated change.

File: .gitignore (hunk 1 above)
Comment (hacksparrow) [inline .gitignore]: Sure.

File: bin/express [hunk 3]
```diff
@@ -185,6 +190,12 @@ function createApplication(app_name, path) {
     // Template support
     app = app.replace('{views}', program.template);
 
+    // Conditionally remove dust.js template code
+    if (program.template !== 'dust') {
+      app = app.replace("var adaro = require('adaro');\n", '');
```
Comment (dougwilson) [inline bin/express]: This is going to be fragile. Can it be done similar to how the CSS stuff is done, where this is added, not removed?

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "express-generator",
   "description": "Express' application generator",
-  "version": "4.13.1",
+  "version": "4.14.0",
```
Comment (dougwilson) [inline package.json]: Please don't bump the version in a pull request.

File: README.md (hunk 2 above)
Comment (hacksparrow) [inline README.md]: Wasn't sure about this actually. Thanks for clarifying.

File: bin/express (hunk 3 above)
Comment (hacksparrow) [inline bin/express]: Will look into it.

File: package.json (hunk 4 above)
Comment (hacksparrow) [inline package.json]: Ok. We should add this fact to the contribution guidelines doc.

/cc @tunniclm

Comment (dougwilson) [inline package.json]: It is common etiquette that a pull request never bumps a version, ever. Not sure why that would need to be spelled out. The pull request is only to add features, etc., not to dictate release schedules. Bumping the version here is missing a lot of things, for example, the version of this module follows the version on Express. I don't see this pull request bumping the Express dep to match the new version, and there are many other things that need to be done in this module before just bumping the version.

Are you not able to remove the version bump from the pull request?

Comment (dougwilson) [inline package.json]: There is not even a 4.14 line of Express yet, so bumping wouldn't even make sense (and version bumps are done in their own commit, when a version is cut @tunniclm already wrote about this in the pending release documentation).

Comment (hacksparrow) [inline package.json]: > Are you not able to remove the version bump from the pull request?

I updated it on my local repo a minute or so after your comment, it's just not pushed.

> There is not even a 4.14 line of Express yet ...

I was not aware about them being needing to be in sync, thanks for letting me know.

**Comment by hacksparrow** [comment]
@dougwilson I have made the changes as per you suggestion.

**Comment by dougwilson** [comment]
Sorry this got closed from some GitHub voodoo. It doesn't have the option for me to re-open it form some reason! :( Please re-create your pull request.

**Comment by hacksparrow** [comment]
Something crazy going on in here! I can see some totally unrelated comments from 2014 and 2015. Might be best not to do anything till the dust settles.

**Comment by dougwilson** [comment]
Re-opened as #121, thanks, @hacksparrow!
